draft-thomson-train-protocol.md

```diff
@@ -336,6 +336,26 @@ same datagram SHOULD ignore any rate limit signal.  Such a datagram might be
 entirely spoofed.
 
 
+## Feedback To Sender About Signals {#feedback}
+
+Information about rate limits is intended for the sending application.  Any
+signal from network elements can be propagated to the receiving application
+using an implementation-defined mechanism.
+
+This document does not define a means for indicating what was received.
+Feedback can be implemented in a manner that best suits applications
+requirements.
```

Comment:
I'm trying to distill the commentary about application-layer signals.  I wonder how much of that needs to be memorialized.

```suggestion
This document does not define a means for indicating what was received.
How a receiving application communicates the rate limit signal to a
sending application will depend on the application in use.

Different applications can choose different approaches. For example,
in an application where a receiver drives rate adaptation, it might
not be necessary to define additional signaling.
```

Reply:
I may be wrong, but I believe a lot of arguments against adding feedback in the network is the "slippery slope". If we automate the transport level feedback, there will be a great temptation to use the advertised rate as part of transport control or congestion control. The slippery part is that if we do that, we create an incentive for network operators to manipulate the advertised rates in order to influence transports, moving away from a pure application mechanism.

Maybe capture your comment "my argument is very specifically about end-to-end".

Reply:
How about:

```suggestion
This document does not define a means for indicating what was received.
That is, the expectation is that any signal is propagated to the application
for handling, not handled automatically by the transport layer.
How a receiving application communicates the rate limit signal to a
sending application will depend on the application in use.

Different applications can choose different approaches. For example,
in an application where a receiver drives rate adaptation, it might
not be necessary to define additional signaling.
```